mixer/pkg/il/compiler/compiler_test.go

```diff
@@ -15,1172 +15,50 @@
-  resolve_f "t1"
-  resolve_f "t2"
-  call timestamp_equal
-  ret
-end`,
-	},
-	{
-		expr: `t1 == t2`,
-		input: map[string]interface{}{
-			"t1": t,
-			"t2": t,
-		},
-		result: true,
-		code: `fn eval() bool
-  resolve_f "t1"
-  resolve_f "t2"
-  call timestamp_equal
-  ret
-end`,
-	},
-	{
-		expr: `t1 != t2`,
-		input: map[string]interface{}{
-			"t1": t,
-			"t2": t,
-		},
-		result: false,
-		code: `fn eval() bool
-  resolve_f "t1"
-  resolve_f "t2"
-  call timestamp_equal
-  not
-  ret
-end`,
-	},
-	{
-		expr: `t1 | t2`,
-		input: map[string]interface{}{
-			"t2": t2,
-		},
-		result: t2,
-	},
-	{
-		expr: `t1 | t2`,
-		input: map[string]interface{}{
-			"t1": t,
-			"t2": t2,
-		},
-		result: t,
-	},
-}
-
-var globalConfig = pb.GlobalConfig{
-	Manifests: []*pb.AttributeManifest{
-		{
-			Attributes: map[string]*pb.AttributeManifest_AttributeInfo{
-				"ai": {
-					ValueType: pbv.INT64,
-				},
-				"ab": {
-					ValueType: pbv.BOOL,
-				},
-				"as": {
-					ValueType: pbv.STRING,
-				},
-				"ad": {
-					ValueType: pbv.DOUBLE,
-				},
-				"ar": {
-					ValueType: pbv.STRING_MAP,
-				},
-				"adur": {
-					ValueType: pbv.DURATION,
-				},
-				"at": {
-					ValueType: pbv.TIMESTAMP,
-				},
-				"aip": {
-					ValueType: pbv.IP_ADDRESS,
-				},
-				"bi": {
-					ValueType: pbv.INT64,
-				},
-				"bb": {
-					ValueType: pbv.BOOL,
-				},
-				"bs": {
-					ValueType: pbv.STRING,
-				},
-				"bd": {
-					ValueType: pbv.DOUBLE,
-				},
-				"br": {
-					ValueType: pbv.STRING_MAP,
-				},
-				"bdur": {
-					ValueType: pbv.DURATION,
-				},
-				"bt": {
-					ValueType: pbv.TIMESTAMP,
-				},
-				"t1": {
-					ValueType: pbv.TIMESTAMP,
-				},
-				"t2": {
-					ValueType: pbv.TIMESTAMP,
-				},
-				"bip": {
-					ValueType: pbv.IP_ADDRESS,
-				},
-				"b1": {
-					ValueType: pbv.BOOL,
-				},
-				"b2": {
-					ValueType: pbv.BOOL,
-				},
-				"sm": {
-					ValueType: pbv.STRING_MAP,
-				},
-			},
-		},
-	},
-}
-
 func TestCompile(t *testing.T) {
 
-	finder := descriptor.NewFinder(&globalConfig)
+	for i, test := range ilt.TestData {
+		if test.E == "" {
+			continue
+		}
 
-	for i, te := range tests {
-		t.Run(fmt.Sprintf("%d '%s'", i, te.expr), func(tt *testing.T) {
-			result, err := Compile(te.expr, finder)
+		conf := test.Conf
+		if conf == nil {
+			conf = ilt.TestConfigs["Default"]
+		}
+		finder := descriptor.NewFinder(conf)
+
+		name := fmt.Sprintf("%d '%s'", i, test.E)
+		t.Run(name, func(tt *testing.T) {
+			result, err := Compile(test.E, finder)
 			if err != nil {
-				tt.Fatalf("error received during compile: %v", err)
+				if err.Error() != test.CompileErr {
+					tt.Fatalf("Unexpected error: '%s' != '%s'", err.Error(), test.CompileErr)
+				}
+				return
 			}
-			actual := text.WriteText(result.Program)
-			if len(te.code) > 0 {
-				if strings.TrimSpace(actual) != strings.TrimSpace(te.code) {
+
+			if test.CompileErr != "" {
+				tt.Fatalf("expected error not found: '%s'", test.CompileErr)
+				return
```

Comment:
probably don't need return after Fatalf

Reply:
Actually, it turns out, I do. :) Without these, the tests do not work as expected.